docs/table-upgrade.md

```diff
@@ -0,0 +1,41 @@
+# Table Upgrade logic and data structures
+The Hive Metastore migration process will upgrade the following Assets:
```

Comment:
can you split this into three different sections in this markdown file? it'll be easier to review/read